WFLY-6974 add IIOP transformers following elytron

ctomc opened the pull request:
also: 
- set subsystem model version to 2.0 (it was skipped for no reason)
- set xsd version to 2.0

ctomc committed:
WFLY-6974 add IIOP transformers following elytron

ctomc commented:
@tadamski can you review / approve this

bstansberry commented on iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPTransformers.java:
```diff
@@ -0,0 +1,111 @@
+
+        ResourceTransformationDescriptionBuilder builder = chained.createBuilder(CURRENT_MODEL_VERSION, VERSION_1);
+        builder.getAttributeBuilder()
+                .addRejectCheck(RejectAttributeChecker.DEFINED, IIOPRootDefinition.SERVER_SSL_CONTEXT, IIOPRootDefinition.CLIENT_SSL_CONTEXT, IIOPRootDefinition.AUTHENTICATION_CONTEXT)
```
Should there be a setDiscard for each of these? There is one for AUTHENTICATION_CONTEXT at L62, but I think it's better if that comes before the addRejectCheck. It's a bit more intuitive "discard x, and then reject !x"

tadamski replied:
+1 
@sguilhen can you please look and confirm?

sguilhen replied:
I think Brian has a good point with the setDiscard. I didn't think of it when I added the transformers a while ago but it does make sense. The rest of the commit gets a +1 from me.

ctomc replied:
fixed

bstansberry commented on iiop-openjdk/src/test/java/org/wildfly/iiop/openjdk/IIOPSubsystemTestCase.java:
```diff
@@ -140,59 +131,6 @@ public void testParseSubsystemWithBadChild() throws Exception {
-        // make sure the models from the two controllers are identical.
-        super.compare(modelA, modelB);
-
-    }
```
Are these dropped tests covered elsewhere?

ctomc replied:
This is all part of AbstractSubsystemBaseTest, but since we had some old guide on how to write subystems, people just copy & pasted that tests into their new subsystems. 
In short it is removed as it is redundant.

bstansberry reviewed (commented):
Thanks for this. When you do these can you separate the refactor to the new ServiceLoader transformer approach into a separate commit from the substantive change. That makes it a lot easier to review the latter.

ctomc commented:
@bstansberry fixed the comments. I do agree that refactoring should be part of separate commit. will make sure that for future work.

ctomc committed:
WFLY-6974 move naming transformers to service loader API

ctomc committed:
WFLY-6974 add naming subsystem transformers

bstansberry commented on iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPTransformers.java:
```diff
@@ -0,0 +1,91 @@
+        Different 'default' for attribute 'trust-in-client'. Current: undefined; legacy: "none"
+        Different 'default' for attribute 'trust-in-target'. Current: undefined; legacy: "none"
+        Missing parameters for operation 'add' in current: []; missing in legacy [server-requires-ssl, server-ssl-context, client-requires-ssl, authentication-context, client-ssl-context]
+         */
```
Nice!

bstansberry commented:
I approved this but Kabir had a comment in the JBEAP variant that looks valid. So, @kabir once you are satisfied this is good.

ctomc committed:
WFLY-6974 refactor weld transformers to non-deprecated api

ctomc commented:
@bstansberry @kabir I've addressed all comments you both had, I've also added few commits that touch transformers in naming and weld subsystems.
But it is mostly transformers api refactor and transformers test tuning

bstansberry commented:
Ok, thanks. The new weld commit looks good.

Please no more commits on this one so it can be merged. :)
